src/library/dao/cue.cpp

```diff
@@ -6,14 +6,19 @@
 
 #include "library/dao/cue.h"
 #include "util/assert.h"
+#include "util/compatibility.h"
 
 namespace {
     const QColor kDefaultColor = QColor("#FF0000");
     const QString kDefaultLabel = ""; // empty string, not null
 }
 
-Cue::~Cue() {
-    //qDebug() << "~Cue()" << m_iId;
+//static
+void CuePointer::deleteLater(Cue* pCue) {
+    if (pCue) {
+        // QObject::deleteLater() is not thread-safe!
```

Comment:
Why do you think that? 
deleteLater() calls QCoreApplication::postEvent() which is thread save.

Reply:
Because deleteLater() is a regular slot and the documentation does not explicitly state that it is thread-safe to be called from other threads? I didn't check the source code. Just reasoned according to the rules of the [Qt5 documentation](http://doc.qt.io/qt-5/threads-reentrancy.html):

"**Note**: Qt classes are only documented as thread-safe if they are intended to be used by multiple threads. If a function is not marked as thread-safe or reentrant, it should not be used from different threads. If a class is not marked as thread-safe or reentrant then a specific instance of that class should not be accessed from different threads."

Reply:
Ok, so you solution is correct and we can go with it.
However it uses two times the Qt meassage queue.
Did you consider to call QCoreApplication::postEvent() directly?